### PR TITLE
[Debt] Update the admin hero

### LINKED ISSUE #8584: ♻️ Update and consolidate the admin-facing Header UI component
image

## ✨ Summary

[Visit the Figma file](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=912-2455&mode=design) • Contact @substrae with design questions

The goal of this ticket is to standardize the header UI element used on admin-facing page. Right now, the implementation is currently inconsistent across pages due to it having evolved over the course of design elements being produced. This component should centralize the styling and children elements so that updates propagate to all pages moving forward.

```[tasklist]
### ✅ Requirements
- [ ] Update the component design to match
- [ ] Add a tab variant using the updated `TabGroup` UI component
- [ ] Ensure the Government of Canada banner and `Breadcrumbs` components are using the same component as the applicant-facing `Header` / `Hero` component, but also make sure that the padding or lack-thereof matches the designs in both cases (they are different)
- [ ] Ensure all header elements are aligned with matching padding and that page content aligns with these elements in all cases
- [ ] Ensure dark mode styles are applied and that Storybook contains light and dark side-by-side comparisons (see other component stories for examples, `Button`, `Hero`, etc.)
- [ ] Ensure all admin pages use this new header in the correct variant
- [ ] Remove all `Breadcrumbs` elements from admin-facing page footers
``` 

### Tab variation

The admin interfaces are unique in that they sometimes require a tabbed interface. [Updates to the tab component](https://github.com/GCTC-NTGC/gc-digital-talent/issues/8582) should block this ticket to ensure design integrity.

<details>
  <summary>📸 With tabs</summary>
  
</details> 

### Dark mode

If you have any questions about styling dark mode, reach out to @substrae. Don't forget that you can use the `:all` Hydrogen modifier to force a style to persist across modes when necessary.

<details>
  <summary>📸 With crumbs</summary>
  
</details>

<details>
  <summary>📸 With tabs</summary>
  
</details>

### Menu states

<details>
  <summary>📸 Menu pinned</summary>
  
</details> 

<details>
  <summary>📸 Menu collapsed</summary>
  
</details> 

### Mobile experience

<details>
  <summary>📸 With crumbs</summary>
  
</details>

<details>
  <summary>📸 With tabs</summary>
  
</details>

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8584 

## 👋 Introduction

This updates the admin hero to be consistent across all pages under `/admin`

## 🕵️ Details

There were 3 variants created:

- Default, no nav
- Sub nav, contains links styled like tabs to sub pages
- Breadcrumbs, contains breadcrumb links

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Navigate to pages under `/admin`
3. Confirm they have a consistent hero

### Default

Screenshot 2023-12-20 134508

### Sub Navigation

Screenshot 2023-12-20 134502

### Breadcrumbs

Screenshot 2023-12-20 134454

### CODE REVIEW

File: apps/web/src/pages/Skills/UpdateSkillPage.tsx
Comment (vd1992) [inline apps/web/src/pages/Skills/UpdateSkillPage.tsx:1]: Crumbs were apparently broken on this page for a long while 🙃 

image

From

image

File: packages/ui/src/components/Tabs/utils.ts
Comment (vd1992) [inline packages/ui/src/components/Tabs/utils.ts:1]: I think you want to keep that padding

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Found some stuff for you to take a look at

File: apps/web/src/pages/SkillFamilies/IndexSkillFamilyPage.tsx [hunk 1]
```diff
@@ -32,11 +31,16 @@ const IndexSkillFamilyPage = () => {
...
-      <PageHeader icon={UserGroupIcon}>{pageTitle}</PageHeader>
-      <SkillFamilyTableApi title={pageTitle} />
-    </AdminContentWrapper>
+      <AdminHero
+        title={pageTitle}
+        nav={{ mode: "subNav", items: navigationCrumbs }}
+      />
+      <AdminContentWrapper>
+        <SkillFamilyTableApi title={pageTitle} />
+      </AdminContentWrapper>
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/SkillFamilies/IndexSkillFamilyPage.tsx]: Oops, yeah that was an early test I forgot to swap back 😅 

Fixed in [fix sub nav mode](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8848/commits/2ecd10d81f869a46503c7833f775a586e4ed4f99)

File: apps/web/src/pages/Skills/UpdateSkillPage.tsx
Comment (esizer) [reply apps/web/src/pages/Skills/UpdateSkillPage.tsx:1]: ooo! Great catch! fixed in [fix skill breadcrumbs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8848/commits/b10cfc9802310623098d7625a3ee3a7c3a21f96f)

File: packages/ui/src/components/Tabs/utils.ts
Comment (esizer) [reply packages/ui/src/components/Tabs/utils.ts:1]: Oh thats what that was doing 😆 . It was causing problems with the nav version. I moved it to the tabs version to only apply it there.

[fix tabs content padding](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8848/commits/15ec8ac4222f0820a3f72374d34e20318be3d51e)

**Review by vd1992 — APPROVED** [review]
Appears all good to me, fantastic!

**Pull request merged**